### PR TITLE
fix(hf): exclude padding tokens from embedding pooling

### PR DESCRIPTION
**Pull request by Shizoqua**
## Description

`hf_embed()` calculates embeddings by averaging `outputs.last_hidden_state` across the complete sequence dimension. This includes hidden states belonging to padding tokens.

Consequently, the embedding produced for the same text can change depending on the amount of padding introduced by other texts in the same batch. The tokenizer already provides an `attention_mask`, and the model receives it, but the pooling step did not use it.

This change applies the attention mask when calculating the mean so that only non-padding tokens contribute to the embedding. The denominator uses `clamp_min(1)` to keep fully masked rows finite and avoid division by zero.

## Related Issues

No related issue.

## Changes Made

- Expand the attention mask across the hidden-state dimension.
- Exclude padding-token hidden states from the pooled embedding.
- Divide by the number of unmasked tokens in each row.
- Protect fully masked rows from division by zero.
- Preserve the existing output shape, dtype, and device.
- Accumulate masked hidden states and token counts in float32 before restoring the original output dtype.
- Add deterministic tests using a NumPy-backed FakeTensor without downloading a model or requiring PyTorch in CI.

## Checklist

- [x] Changes tested locally
- [x] Code reviewed
- [x] Documentation updated (not necessary for this fix)
- [x] Unit tests added

## Test Results

- New masked-pooling tests: 8 passed
- Fails-before behaviour reproduced on unmodified `main`
- Unpadded input behaviour remains unchanged
- Fully masked rows remain finite
- `pre-commit run --all-files`: passed
- `git diff --check`: passed

## Additional Notes

The regression test demonstrates that the previous pooling calculation changes the embedding for identical token content when additional padding is present. With attention-mask-weighted pooling, the embedding remains stable regardless of batch padding.

### CODE REVIEW

**Comment by danielaskdd** [comment]
@codex review

File: lightrag/llm/hf.py [hunk 1]
```diff
@@ -219,11 +219,19 @@ async def hf_embed(
...
+            attention_mask=attention_mask,
         )
-        embeddings = outputs.last_hidden_state.mean(dim=1)
+        # Plain .mean(dim=1) counts padding-token hidden states, so the same
+        # text's embedding shifts depending on what else is in the batch.
+        # Weight by attention_mask instead; clamp_min(1) keeps a fully-masked
+        # row finite (all-padding input) rather than dividing by zero.
+        mask = attention_mask.unsqueeze(-1).to(outputs.last_hidden_state.dtype)
+        summed = (outputs.last_hidden_state * mask).sum(dim=1)
+        counts = mask.sum(dim=1).clamp_min(1)
```
[the remark this answers is not in the document]
Comment (Shizoqua) [reply lightrag/llm/hf.py]: Fixed in 3820e48. The pooling reductions now use float32, with regression tests added. All CI checks pass.

**Comment by danielaskdd** [comment]
@codex review
